sklearn/linear_model/ridge.py

```diff
@@ -141,7 +144,14 @@ def _solve_cholesky_kernel(K, y, alpha, sample_weight=None):
         # Only one penalty, we can solve multi-target problems in one time.
         K.flat[::n_samples + 1] += alpha[0]
 
-        dual_coef = linalg.solve(K, y, sym_pos=True, overwrite_a=True)
+        try:
+            # XXX: Check if overwrite_a=True is ok when copy is False
```

Comment:
I think, we have to change that to overwrite_a=False. In case that a LinAlgError is raised and the pseudo-inverse of K is computed instead, the pseudo-inverse would be based on the modified K.